specification/loadtestservice/data-plane/Microsoft.LoadTestService/preview/2021-07-01-preview/loadtestservice.json

```diff
@@ -0,0 +1,2639 @@
+            "default": 50
+          },
+          {
+            "in": "query",
+            "name": "api-version",
+            "description": "Api version",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The requested loadtests",
+            "schema": {
+              "$ref": "#/definitions/TestModelResourceList"
+            }
+          },
+          "default": {
+            "description": "*** Error Responses: ***\n\n * \"400\", \"InvalidHostOrResourceName\", \"Host name or domain is not valid\"\n\n* \"500\", \"InternalServerError\", \"Unable to process the request, please retry\"",
+            "schema": {
+              "$ref": "#/definitions/DefaultErrorResponseBody"
+            },
+            "headers": {
+              "x-ms-error-code": {
+                "description": "The error code for specific error that occurred.",
+                "type": "string"
+              }
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Test_ListLoadTestSearch_MaximumSet_Gen": {
+            "$ref": "./examples/Test_ListLoadTestSearch_MaximumSet_Gen.json"
+          },
+          "Test_ListLoadTestSearch_MinimumSet_Gen": {
+            "$ref": "./examples/Test_ListLoadTestSearch_MinimumSet_Gen.json"
+          }
+        }
+      }
+    },
+    "/loadtests/{testId}/files/{fileId}": {
+      "put": {
+        "tags": [
+          "Test"
+        ],
+        "summary": "Upload input file for a given test name. File name must be a valid URL character ^[a-z0-9_-]*$. File size can't be more than 50 MB. Existing file with same name for the given test will be overwritten.",
+        "operationId": "Test_UploadTestFile",
+        "consumes": [
+          "multipart/form-data"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "path",
+            "name": "testId",
+            "description": "Unique identifier for load test name, must be a valid URL character ^[a-z0-9_-]*$",
+            "required": true,
+            "type": "string",
+            "maxLength": 50,
+            "minLength": 2,
+            "pattern": "^[a-z0-9_-]*$"
+          },
+          {
+            "in": "path",
+            "name": "fileId",
+            "description": "Unique identifier for test file, must be a valid URL character ^[a-z0-9_-]*$",
+            "required": true,
+            "type": "string",
+            "maxLength": 50,
+            "minLength": 2,
+            "pattern": "^[a-z0-9_-]*$"
+          },
+          {
+            "in": "query",
+            "name": "api-version",
+            "description": "Api version",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "in": "formData",
+            "name": "file",
+            "required": true,
+            "type": "file"
+          }
```

Comment:
As above, this should be a "body" parameter with content-type "application/octet-stream".

Reply:
We will change in our next API revision